src/cpu/cpu_inner_product_list.cpp

```diff
@@ -69,6 +69,11 @@ const std::map<pk_dt_impl_key_t, std::vector<impl_list_item_t>> &impl_list_map()
             CPU_INSTANCE_AVX2(brgemm_inner_product_fwd_t, avx2)
             nullptr,
         }},
+        {{forward, f32, f16, f32}, {
+            CPU_INSTANCE_AVX512(brgemm_inner_product_fwd_t, avx512_core)
+            CPU_INSTANCE_AVX2(brgemm_inner_product_fwd_t, avx2)
+            nullptr,
+        }},
```

Comment:
Could you, please, provide a use case for such configuration?

Reply:
Hey @dzarukin. This is needed for LLMs with fp16 compressed weights (lower speedup than for u8/u4 compressed weights, but no accuracy loss comparing with fp32).

Reply:
@dmitry-gorokhov thanks for explaining. Should oneDNN follow the same path and add such configurations support to the library?

Reply:
@dzarukin I would highly recommend to do that. Pretty cheap optimization in terms of implementation effort, but very benefitial for LLMs at the same time. Should be useful for majority of DL frameworks focusing on edge inference, since weights conversion to FP16 doesn't require any fine-tuning/complex compression techniques,